### PR TITLE
Certificate server

### PR DESCRIPTION
**Pull request by jana-rapava**

### CODE REVIEW

File: src/certgen_certificate_config.cc [hunk 1]
```diff
@@ -66,7 +66,22 @@ certgen_certificate_config_test (bool verbose)
     printf (" * certgen_certificate_config: ");
 
     //  @selftest
-    //  Simple create/destroy test
+    //  Simple deserialize test
+    std::string configFilePath (SELFTEST_DIR_RO + std::string ("certificate.config"));
```
Comment (perrettecl) [inline src/certgen_certificate_config.cc]: Test need to be performed with at least one file well formed and one file formed correctly (positive vs negative tests)

File: src/certgen_certificate_generator_server.cc [hunk 2]
```diff
@@ -32,6 +32,75 @@
 
 namespace certgen
 {
+    using namespace fty;
+    using namespace std::placeholders;
+
+    CertificateGeneratorServer::CertificateGeneratorServer()
+    {
+        //initiate the commands handlers
+        m_supportedCommands[GENERATE_SELFSIGNED_CERTIFICATE] = std::bind(&CertificateGeneratorServer::handleGenerateSelfsignedCertificate, this, _1, _2);
```
Comment (perrettecl) [inline src/certgen_certificate_generator_server.cc]: Sender is not going to be necessary

File: src/certgen_certificate_generator_server.cc [hunk 3]
```diff
@@ -32,6 +32,75 @@
...
+            return {"ERROR", e.what()};
+        }
+        catch (...) //show must go one => Log and ignore the unknown error
+        {
+            log_error("Unexpected error: unknown");
+            return {"ERROR", ""};
+        }
+    }
+
+    std::string CertificateGeneratorServer::handleGenerateSelfsignedCertificate(const fty::Sender & /*sender*/, const fty::Payload & /*params*/)
```
Comment (perrettecl) [inline src/certgen_certificate_generator_server.cc]: Sender is not going to be necessary, we can remove it from the function

**Review by perrettecl — CHANGES_REQUESTED** [review]
Project.xml need to be updated

**Review by stark-dev — APPROVED** [review]

**Pull request merged**